### PR TITLE
Memory-integration

### PR DESCRIPTION
**Pull request by Tyler-R-Kendrick**
added basic memory integration for few shot examples

### CODE REVIEW

File: src/Web/FunctionFilters/CodeRetryFunctionFilter.cs [hunk 1]
```diff
@@ -1,98 +1,70 @@
...
-    private Kernel _kernel;
+  private readonly Dictionary<string, int> _numRetries = [];
 
-    private const int MAX_RETRIES = 3;
+  private const int MAX_RETRIES = 3;
 
-    public CodeRetryFunctionFilter(ChatSessionService chatSessionService, Kernel kernel)
+  public async void OnFunctionInvoked(FunctionInvokedContext context)
+  {
+    if (context.Function.Name != "CodeGen")
```
[the remark this answers is not in the document]
Comment (Tyler-R-Kendrick) [reply src/Web/FunctionFilters/CodeRetryFunctionFilter.cs:14]: I actually missed this. This won't work. It needs to be the registered name of the function - which is a magic string.
I have a different pattern of registering the function filter when registering a function, so you don't have magic strings.
That requires a bit of extra code that I'm discussing with the Semantic Kernel team to implement in their API Design.

I don't have qualms pulling it in here to make the code more rigorous; but I want to make sure I don't obfuscate with complexity - I have a bad habit of that.

File: src/Web/FunctionFilters/CodeRetryFunctionFilter.cs [hunk 2]
```diff
@@ -1,98 +1,70 @@
...
-        else
+        chatSession.AddUserMessage("Please correct the errors and try again.");
+        context.Arguments["history"] = chatSession;
+        // re-invoke the function
+        var nextResult = await context.Function.InvokeAsync(kernel, context.Arguments);
+        context.SetResultValue(nextResult.GetValue<string>());
+      }
+      else
+      {
+        var errorResponse = $"{code}{Environment.NewLine}I'm sorry, but the code I could generate contains the following errors.  You'll need to correct them before using this code.";
```
[the remark this answers is not in the document]
Comment (Tyler-R-Kendrick) [reply src/Web/FunctionFilters/CodeRetryFunctionFilter.cs:59]: That would be more robust, but there's aren't responses shared with the user - these are what gets passed to the context window of the LLM.

File: src/Web/Services/LanguageService.cs [hunk 3]
```diff
@@ -1,122 +1,101 @@
...
+    var value = blobServiceOptions.Value;
+    Guard.IsNotNull(value.AccountName, nameof(value.AccountName));
+    Guard.IsNotNull(value.AccessKey, nameof(value.AccessKey));
+    Guard.IsNotNull(value.ContainerName, nameof(value.ContainerName));
+
+    StorageSharedKeyCredential storageSharedKeyCredential =
+      new(value.AccountName,
+        value.AccessKey);
+    string blobServiceEndpoint = $"https://{value.AccountName}.blob.core.windows.net";
+    _blobServiceClient = new(new(blobServiceEndpoint), storageSharedKeyCredential);
```
[the remark this answers is not in the document]
Comment (Tyler-R-Kendrick) [reply src/Web/Services/LanguageService.cs:28]: I agree entirely. All new invocations should be done through a factory or DI.

**Review by jaspecla — APPROVED** [review]
This has been sufficiently scrutinized, so in light of that existing scrutiny:  LGTM

**Review by petemessina — APPROVED** [review]
LGTM

**Pull request merged**